### PR TITLE
refactor(version pages): migrate confirm to modal

### PR DESCRIPTION
**Pull request by aeng**
https://zanata.atlassian.net/browse/ZNTA-1285

---
This change is [Reviewable](https://reviewable.io/reviews/zanata/zanata-server/1266)

### CODE REVIEW

**Comment by huangp** [comment]
Review status: 0 of 8 files reviewed at latest revision, 1 unresolved discussion, some commit checks failed.

---

*[zanata-war/src/main/java/org/zanata/action/VersionHomeAction.java, line 636 \[r1\]](https://reviewable.io:443/reviews/zanata/zanata-server/1266#-KOw3wqattXQ-rliZxHp:-KOw3wqattXQ-rliZxHq:bltsczl) ([raw file](https://github.com/zanata/zanata-server/blob/3c874d6758cb31022b1b3e755eb5e1cf333b3a60/zanata-war/src/main/java/org/zanata/action/VersionHomeAction.java#L636)):*
> ```Java
>     public void deleteDocument(String docId) {
>         checkDocumentRemovalAllowed();
>         HDocument doc = documentDAO.getById(new Long(docId));
> ```

I think Long.valueOf(docId) will be better

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/zanata/zanata-server/1266)*

**Comment by huangp** [comment]
:lgtm: -(Awaiting review)  +Reviewed +(On QA) 

---

Reviewed 7 of 8 files at r1, 1 of 1 files at r2.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/zanata/zanata-server/1266#-:-KOwBVx8vaBw4PkaYLg8:b-nkuomy)*

File: zanata-war/src/main/webapp/iteration/view.xhtml [hunk 1]
```diff
@@ -127,10 +127,54 @@
...
+    <script type="application/javascript">
+      function onExportAllTMXResponse(exp) {
+        if (exp) {
+          window.open("#{request.contextPath}/rest/tm/projects/#{versionHomeAction.projectSlug}/iterations/#{versionHomeAction.versionSlug}");
+        }
+        zanata.modal.hide('#exportTMXVersionModal');
+      }
+    </script>
+    <zanata:confirm-modal id="exportTMXVersionModal"
+      title="#{msgs['jsf.ExportTMXIter']}"
```
Comment (djansen-redhat) [inline zanata-war/src/main/webapp/iteration/view.xhtml]: Does not exist in messages.properties. There's a jsf.ExportTMX (which could potentially be renamed but I don't know what the impact would be)

File: zanata-war/src/main/webapp/WEB-INF/layout/version/languages-tab.xhtml [hunk 2]
```diff
@@ -68,6 +68,31 @@
   </div>
 
   <div style="position: relative">
+
+    <ui:fragment
+      rendered="#{versionHomeAction.zipFileDownloadAllowed}">
```
Comment (djansen-redhat) [inline zanata-war/src/main/webapp/WEB-INF/layout/version/languages-tab.xhtml]: Is this the right permission check?

**Comment by djansen-redhat** [comment]
Weirdness with download all translations dialog
downloadall

**Comment by djansen-redhat** [comment]
And with the language tmx download dialog
downloadlangtmx

File: zanata-war/src/main/webapp/WEB-INF/layout/version/languages-tab.xhtml (hunk 2 above)
Comment (aeng) [inline zanata-war/src/main/webapp/WEB-INF/layout/version/languages-tab.xhtml]: Thats the one used before any changes. So its correct.

File: zanata-war/src/main/webapp/iteration/view.xhtml (hunk 1 above)
Comment (aeng) [inline zanata-war/src/main/webapp/iteration/view.xhtml]: Good catch.

**Comment by djansen-redhat** [comment]
Cancel CopyTrans dialog has title "cc.attrs.cancelTooltip"

**Comment by aeng** [comment]
@djansen-redhat fixed the tooltip.

**Comment by djansen-redhat** [comment]
:white_check_mark: IT'S DONE

**Comment by aeng** [comment]
@zanata-jenkins retest this please

**Pull request merged**